internal/localdebug/local_ide_debug.go

```diff
@@ -0,0 +1,286 @@
+package localdebug
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/alecthomas/types/optional"
+
+	"github.com/TBD54566975/ftl/internal/log"
+)
+
+const projectToml = "project.toml"
+const jvmDebugConfig = `<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="%s" type="Remote">
+    <option name="USE_SOCKET_TRANSPORT" value="true" />
+    <option name="SERVER_MODE" value="false" />
+    <option name="SHMEM_ADDRESS" />
+    <option name="HOST" value="localhost" />
+    <option name="PORT" value="%d" />
+    <option name="AUTO_RESTART" value="false" />
+    <RunnerSettings RunnerId="Debug">
+      <option name="DEBUG_PORT" value="%d" />
+      <option name="LOCAL" value="false" />
+    </RunnerSettings>
+    <method v="2" />
+  </configuration>
+</component>`
+
+const golangDebugConfig = `<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="%s" type="GoRemoteDebugConfigurationType" factoryName="Go Remote" port="%d">
+    <option name="disconnectOption" value="LEAVE" />
+    <disconnect value="LEAVE" />
+    <method v="2" />
+  </configuration>
+</component>`
+
+type IDEIntegration struct {
+	cxt         context.Context
+	projectPath string
+}
+
+type DebugInfo struct {
+	Port     int
+	Language string
+}
+type contextKeyType struct{}
+
+var contextKey = contextKeyType{}
+
+func NewIDEIntegration(cxt context.Context, projectPath string) *IDEIntegration {
+	ret := &IDEIntegration{cxt: cxt, projectPath: projectPath}
+	return ret
+}
+
+func (r *IDEIntegration) IntoContext(ctx context.Context) context.Context {
+	return context.WithValue(ctx, contextKey, r)
+}
+
+func FromContext(ctx context.Context) optional.Option[IDEIntegration] {
+	value := ctx.Value(contextKey)
+	if ide, ok := value.(*IDEIntegration); ok {
+		return optional.Ptr(ide)
+	}
+	return optional.None[IDEIntegration]()
+}
+
+// SyncIDEDebugIntegrations will sync the local IDE debug configurations for the given project path.
+// This is a bit of a hack to prove out the concept of local debugging.
+func (r *IDEIntegration) SyncIDEDebugIntegrations(ports map[string]*DebugInfo) {
+	if r.projectPath == "" {
+		return
+	}
+	r.handleIntellij(ports)
+	r.handleVSCode(ports)
+}
+
+// GetExistingDebugPort will return the existing debug ports for the given project name
+// based on VSCode configurations.
+// VSCode configurations are always created, so there is no need to look at intellij
```

Comment:
Because I don't think vscode creates a .vscode directory by default, only if it actively needs to store stuff. I could create it only if intellij does not exist?

Reply:
Mmm that's a good point, maybe we need some better mechanism for detecting it, or something explicit? Let's get it in and worry about it later.